src/Native/Unix/CMakeLists.txt

```diff
@@ -23,6 +23,10 @@ add_compile_options(-I${CMAKE_CURRENT_SOURCE_DIR}/Common)
 add_compile_options(-I${CMAKE_CURRENT_BINARY_DIR}/Common)
 add_compile_options(-g)
 if(CMAKE_C_COMPILER_ID STREQUAL Clang)
+    if (CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
```

Comment:
I am not a fan of checking the compiler version. It would be better to check if the compiler supports this flag at all. But I dont see us doing this for any other compiler flag.